ast/expr.go

```diff
@@ -668,28 +665,25 @@ func parseOpen(node *syntax.ObjectNode, name *StringExpr, args Expr) (Expr, synt
 		diags.Extend(ExprError(obj, "missing provider inputs ('inputs')"))
 	}
 
-	if diags.HasErrors() {
-		return nil, diags
-	}
-
 	return OpenSyntax(node, name, obj, provider, inputs), diags
 }
 
 func parseShortOpen(node *syntax.ObjectNode, name *StringExpr, args Expr) (Expr, syntax.Diagnostics) {
 	kvp := node.Index(0)
-	provider := strings.TrimPrefix(kvp.Key.Value(), "fn::open::")
+	provider := StringSyntaxValue(name.Syntax().(*syntax.StringNode), strings.TrimPrefix(kvp.Key.Value(), "fn::open::"))
```

Comment:
Is this type assertion safe, or should it be checked?

Reply:
It is safe. If we get to this point, we know we have an underlying string node.